pkgs/development/ruby-modules/gem-config/default.nix

```diff
@@ -78,6 +78,13 @@ in
         [ darwin.apple_sdk.frameworks.CoreServices ];
   };
 
+  jekyll = attrs: {
+    postInstall = ''
+      installPath=$(cat $out/nix-support/gem-meta/install-path)
+      sed -i $installPath/lib/jekyll/commands/new.rb -e 's@Exec.run("bundle", "install"@Exec.run("nix-shell", "-p", "bundler", "--pure", "--run", "${bundler}/bin/bundle install"@'
```

Comment:
For example user in #20617 needs an additional Gem and there is no straightforward way to add it.